src/twister2/builder/cmake_builder.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import logging
+import shutil
+import subprocess
+
+from twister2.builder.builder_abstract import BuilderAbstract
+from twister2.exceptions import TwisterBuildException
+from twister2.helper import log_command
+
+logger = logging.getLogger(__name__)
+
+
+class CmakeBuilder(BuilderAbstract):
+    """Build source code using `cmake`"""
+
+    def build(self, cmake_helper: bool = False) -> None:
+        """
+        Build Zephyr application with `cmake`.
+        """
+        logger.info('Building Zephyr application')
+        self.run_cmake(cmake_helper)
+        if cmake_helper:
+            return
+        self.run_build_generator()
+
+    def run_cmake(self, cmake_helper: bool = False) -> None:
+        cmake = self._get_cmake()
+
+        command = [
+            cmake,
+            f'-S{self.build_config.source_dir}',
+            f'-B{self.build_config.build_dir}',
+            f'-DBOARD={self.build_config.platform_name}'
+        ]
+
+        if cmake_args := self._prepare_cmake_args(self.build_config):
+            command.extend(cmake_args)
+
+        if cmake_helper:
+            command.extend(
+                [
+                    '-DMODULES=dts,kconfig',
+                    f'-P{self.build_config.zephyr_base}/cmake/package_helper.cmake',
+                ]
+            )
+
+        log_command(logger, 'Build Cmake command', command, level=logging.INFO)
+        self._run_command_in_subprocess(command)
+
+    def run_build_generator(self) -> None:
+        cmake = self._get_cmake()
+        command: list[str] = [cmake, '--build', str(self.build_config.build_dir)]
+        log_command(logger, 'Build Cmake command', command, level=logging.INFO)
+        self._run_command_in_subprocess(command)
+
+    def _run_command_in_subprocess(self, command: list[str]) -> None:
```

Comment:
To be more precise in logs and to have still generic method for run command in subprocess I propose to add method argument `action` which can be used next in logs:

```suggestion
    def _run_command_in_subprocess(self, command: list[str], action: str) -> None:
        try:
            process = subprocess.run(
                command,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
            )
        except subprocess.CalledProcessError as e:
            logger.exception(
                'An exception has been raised for %s: %s for %s',
                action, self.build_config.source_dir, self.build_config.platform_name
            )
            raise TwisterBuildException(f'{action} error') from e
        else:
            if process.returncode == 0:
                self._log_output(process.stdout, logging.DEBUG)
                logger.info(
                    'Finished running %s on %s for %s',
                    action, self.build_config.source_dir, self.build_config.platform_name
                )
            else:
                self._log_output(process.stdout, logging.INFO)
                msg = (
                    f'Failed running {action} {self.build_config.source_dir} '
                    f'for platform: {self.build_config.platform_name}'
                )
                logger.error(msg)
                raise TwisterBuildException(msg)
```

Then, thanks to this it can be used in `run_cmake` method it this way:
```diff
- self._run_command_in_subprocess(command)
+ self._run_command_in_subprocess(command, 'CMake')
```
and in `run_build_generator` method it this way:
```diff
- self._run_command_in_subprocess(command)
+ self._run_command_in_subprocess(command, 'building')
```